pkg/i2gw/providers/common/resource_reader.go

```diff
@@ -53,9 +53,18 @@ func ReadIngressesFromCluster(ctx context.Context, client client.Client, ingress
 }
 
 func ReadIngressesFromFile(filename, namespace string, ingressClasses sets.Set[string]) (map[types.NamespacedName]*networkingv1.Ingress, error) {
-	stream, err := os.ReadFile(filename)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read file %v: %w", filename, err)
+	var stream []byte
+	var err error
+	if filename == "-" {
```

Comment:
it is a nit, so feel free to disregard.

As you are repeating this here and on ReadFromServices, maybe a util function could be created to not repeat yourself that much

But lgtm